test/plots/plotTests.ts

```diff
@@ -1,584 +1,302 @@
+          const expectedClipPathURL = "url(" + expectedPrefix + "#" + clipPathId + ")";
+
+          window.history.replaceState(originalState, originalTitle, originalLocation);
+
+          const normalizeClipPath = (s: string) => s.replace(/"/g, "");
+          assert.strictEqual(normalizeClipPath((<any> plot)._element.attr("clip-path")), expectedClipPathURL,
+            "the clipPath reference was updated");
+          svg.remove();
+        });
+      }
+    });
 
-        let svg = TestMethods.generateSVG();
-        let plot = new Plottable.Plot();
-        plot.renderTo(svg);
+    describe("setting the attribute of plot elements", () => {
+      let plot: Plottable.Plot;
+
+      beforeEach(() => {
+        plot = new Plottable.Plot();
+      });
+
+      it("can set the attribute to a constant value", () => {
+        const constantNumber = 10;
+        assert.strictEqual(plot.attr("foo", constantNumber), plot, "setting the attribute returns the calling plot");
+        assert.strictEqual(plot.attr("foo").accessor(null, 0, null), constantNumber, "can set the attribute to a constant number");
+
+        const constantString = "one";
+        plot.attr("foo", constantString);
+        assert.strictEqual(plot.attr("foo").accessor(null, 0, null), constantString, "can set the attribute to a constant string");
+      });
+
+      it("can set the attribute based on the input data", () => {
+        const data = [1, 2, 3, 4, 5];
+        const dataset = new Plottable.Dataset(data);
+        plot.addDataset(dataset);
+
+        const numberAccessor = (d: any, i: number) => d + i * 10;
+        assert.strictEqual(plot.attr("foo", numberAccessor), plot, "setting the attribute returns the calling plot");
+
+        let attrAccessor = plot.attr("foo").accessor;
+        plot.datasets().forEach((dataset, datasetIndex) => {
+          dataset.data().forEach((datum, index) => {
+            assert.strictEqual(attrAccessor(datum, index, dataset), numberAccessor(datum, index),
+              `can set attribute for number datum ${index} in dataset ${datasetIndex}`);
+          });
+        });
+
+        const stringAccessor = (d: any, i: number) => `${d + i * 10} foo`;
+        assert.strictEqual(plot.attr("foo", stringAccessor), plot, "setting the attribute returns the calling plot");
+
+        attrAccessor = plot.attr("foo").accessor;
+        plot.datasets().forEach((dataset, datasetIndex) => {
+          dataset.data().forEach((datum, index) => {
+            assert.strictEqual(attrAccessor(datum, index, dataset), stringAccessor(datum, index),
+              `can set attribute for string datum ${index} in dataset ${datasetIndex}`);
+          });
+        });
+      });
+
+      it("can set the attribute based on the scaled input data", () => {
+        const data = [1, 2, 3, 4, 5];
+        const dataset = new Plottable.Dataset(data);
+        plot.addDataset(dataset);
+
+        const numberAccessor = (d: any, i: number) => d + i * 10;
```

Comment:
That should go in a separate test, right? With an `Accessor` that includes `assert` statements.